Ignore regular writes by cupsd to its own files

petterreinholdtsen opened the pull request:
The CUPS daemon update its /etc/cups/subscriptions.conf file regularly, no need to create noise in the logs about this.

/kind bug
/kind cleanup
/kind feature
/area rules

poiana commented:
@petterreinholdtsen: The label(s) `kind/bug, kind/cleanup, kind/feature, area/rules` cannot be applied, because the repository doesn't have them.

<details>

In response to [this](https://github.com/falcosecurity/rules/pull/24):

>The CUPS daemon update its /etc/cups/subscriptions.conf file regularly, no need to create noise in the logs about this.
>
>/kind bug
>/kind cleanup
>/kind feature
>/area rules


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

poiana commented:
Thanks for your pull request. Before we can look at it, you'll need to add a 'DCO signoff' to your commits.

:memo: **Please follow instructions in the [contributing guide](https://github.com/falcosecurity/.github/blob/main/CONTRIBUTING.md) to update your commits with the DCO**

Full details of the Developer Certificate of Origin can be found at [developercertificate.org](https://developercertificate.org/).

**The list of commits missing DCO signoff**:

- [134d33d](https://github.com/falcosecurity/rules/commits/134d33d5182851ad8ff38e9ef43c5c60bd17507d) Ignore regular writes by cupsd to its own files

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

poiana commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/falcosecurity/rules/pull/24#" title="Author self-approved">petterreinholdtsen</a>*
**Once this PR has been reviewed and has the lgtm label**, please assign mstemm for approval by writing `/assign @mstemm` in a comment. For more information see [the Kubernetes Code Review Process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process).

The full list of commands accepted by this bot can be found [here](https://prow.falco.org/command-help?repo=falcosecurity%2Frules).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/falcosecurity/rules/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["mstemm"]} -->

loresuso commented on rules/falco_rules.yaml:
```diff
@@ -1172,6 +1172,23 @@
+  condition: >
+    proc.name = cupsd and
+    fd.name in (
+       /etc/cups/subscriptions.conf,
+       /etc/cups/subscriptions.conf.N,
+       /etc/cups/subscriptions.conf.O,
+       /etc/cups/printers.conf,
+       /etc/cups/printers.conf.N,
+       /etc/cups/printers.conf.O
```
Hello @petterreinholdtsen, I am not familiar with cupsd but just wanted to ask a question, are we sure that these are the only file it is writing? After installation, I am seeing in `/etc/cups` that there are also other files and directories. 
Maybe we can con consider using `startswith /etc/cups` instead of `in`. Just a question, in any case sgtm

petterreinholdtsen commented:
[Lorenzo Susini]
> Hello @petterreinholdtsen, I am not familiar with cupsd but just
> wanted to ask a question, are we sure that these are the only file it
> is writing? After installation, I am seeing in `/etc/cups` that there
> are also other files and directories.

The files listed are the ones that I am aware of are automatically
updated without human interaction.  Other files (as well as these) are
updated by cupsd when printers and settings are configured, but I
suspect a log entry might be in order when that happen?

> Maybe we can con consider using `startswith /etc/cups` instead of
> `in`. Just a question, in any case sgtm

It depend on how fine grained the rule should be.  Everything under
/etc/cups/ "belong" to cupsd, so it would not be surprising that it
updated one of its own files.

-- 
Happy hacking
Petter Reinholdtsen

loresuso commented:
Since this rule has been noisy historically, I'd use `startswith`. Moreover, I see no security implications and as you said "not surprising" cupsd writing to its own files. This is my opinion, let's wait for other people to join the conversation

petterreinholdtsen commented:
[Lorenzo Susini]
> Since this rule has been noisy historically, I'd use
> `startswith`. Moreover, I see no security implications and as you said
> "not surprising" cupsd writing to its own files. This is my opinion,
> let's wait for other people to join the conversation

I have no problem with a less specific test.  Patch updated to use
fd.name startswith /etc/cups/ instead.

-- 
Happy hacking
Petter Reinholdtsen

darryk10 commented on rules/falco_rules.yaml:
```diff
@@ -1172,6 +1172,17 @@
 
+# Return always_true from this macro to get warnings about the cups daemons
+# regular updates in /etc/cups/.
+- macro: cupsd_state_updates
```
I'm not sure if this macro is actually useful. In the ruleset we have already never_true macros and I would avoid adding a new one if not strictly required.

petterreinholdtsen commented:
[schie]
> I'm not sure if this macro is actually useful. In the ruleset we have
> already never_true macros and I would avoid adding a new one if not
> strictly required.

The macro is there to allow a site to control the behaviour using local
overrides.

-- 
Vennlig hilsen
Petter Reinholdtsen

leogr commented:
/milestone falco-rules-0.2

leogr commented:
/kind feature

leogr commented:
/area rules

poiana commented:
Issues go stale after 90d of inactivity.

Mark the issue as fresh with `/remove-lifecycle stale`.

Stale issues rot after an additional 30d of inactivity and eventually close.

If this issue is safe to close now please do so with `/close`.

Provide feedback via https://github.com/falcosecurity/community.

/lifecycle stale

poiana commented:
Rotten issues close after 30d of inactivity.

Reopen the issue with `/reopen`.

Mark the issue as fresh with `/remove-lifecycle rotten`.

Provide feedback via https://github.com/falcosecurity/community.
/close

poiana commented:
@poiana: Closed this PR.

<details>

In response to [this](https://github.com/falcosecurity/rules/pull/24#issuecomment-1684944716):

>Rotten issues close after 30d of inactivity.
>
>Reopen the issue with `/reopen`.
>
>Mark the issue as fresh with `/remove-lifecycle rotten`.
>
>Provide feedback via https://github.com/falcosecurity/community.
>/close


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

petterreinholdtsen commented:
It is unclear to me what more I can do to make this acceptable into the
default rule set.  Should I drop the option to allow the override and
always ignore changes in /etc/cups/?

I have rebased the change on top of the current main branch.

-- 
Happy hacking
Petter Reinholdtsen
